contracts/cw20-stake-external-rewards/src/contract.rs

```diff
@@ -463,6 +544,32 @@ pub fn execute_update_manager(
         ))
 }
 
+pub fn execute_update_pause_status(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    is_paused: bool,
+) -> Result<Response<Empty>, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+    if Some(info.sender.clone()) != config.owner && Some(info.sender) != config.manager {
+        return Err(ContractError::Unauthorized {});
+    };
+
+    let reward_config = REWARD_CONFIG.load(deps.storage)?;
+    if reward_config.period_finish < env.block.height {
+        return Err(RewardPeriodFinished {});
+    } else if config.is_paused == is_paused {
+        return Err(ContractError::SettingCurrentPauseState {});
```

Comment:
hmm, I would argue that if they are trying to set `is_paused` to what it already is, it should be a no-op (ie no error)

Reply:
I kind of followed the pattern I saw on prop-single:
https://github.com/DA0-DA0/dao-contracts/blob/693a5d6a36ada1505c5a52e8029b1b80ea557393/contracts/cw-proposal-single/src/contract.rs#L339-L343

No strong preference though, what do others think?